### ISSUE TITLE
Expanding/collapsing tabs causes overflow/resize issues

### ISSUE THREAD

**Issue by jwueller** [zen-browser/desktop#2931]
### Captchas

- [X] I have read the instructions.
- [X] I have searched existing issues and avoided creating duplicates.
- [X] I am not filing an enhancement request.

### What happened?

Here is a video of the behavior:
https://github.com/user-attachments/assets/0dbf8df5-3815-4173-b941-8cf7f304fe4b

I can see several issues here:
1. During the expansion animation, the tab list gets shorter, causing a scroll position to shift.
2. After the expansion animation finishes, the tab list goes back to its regular height, but the exact scroll position is already lost.
3. During contraction, the entire UI seems to get taller by some (the same?) amount, enlarging the entire UI to the point that elements get cut off at the bottom of the window. This also causes the document viewport to resize.
4. After contraction, everything goes back to its natural size again.
5. The workspace icon/title disappears during any state of expansion.

Basically, something seems to appear/disappear during the animation, which causes the resizing.

No mods enabled.

#2033 seems similar, but I assume that this might be a regression, since that one was closed as completed.

### Reproducible?

- [X] I have checked that this issue cannot be reproduced on Mozilla Firefox.

### Version

1.0.1-a.19

### What platform are you seeing the problem on?

Linux

**Comment by Shockshwat** [comment]
I must also add that there is no closing animation of the sidebar like the opening animation. It just snaps back.

**Comment by jwueller** [comment]
Note that this also happens when manually expanding/collapsing tabs, so I'm updating the title to match.
